src/ml/neural_net/compute_context.hpp

```diff
@@ -34,19 +38,25 @@ class compute_context {
   class registration {
   public:
     // Registers `factory_fn` at the given priority.
-    registration(int priority, factory factory_fn);
+   registration(int priority, factory factory_fn, tf_factory tf_factory_fn_);
+
+   // Removes the registration. In practice, simplest just not to deallocate...
+   ~registration();
 
-    // Removes the registration. In practice, simplest just not to deallocate...
-    ~registration();
+   int priority() const { return priority_; }
```

Comment:
Did `clang-format` do this, give this an overall indent level of 3 spaces? If so, try giving the `public:` above an indent level of 1 space. I think `clang-format` thinks that stuff should be indented by 1 space relative to `public:`

In short, the number of leading spaces for code should just about always be a multiple of 2, except for `public:`

(See https://google.github.io/styleguide/cppguide.html#Class_Format)